### PR TITLE
Deleted all code referencing maps

### PR DESCRIPTION
**Pull request by mhutchinson**
This deletes the lion's share of the old DB-backed map code. There are still places that will require cleanup like database schemas, but this is a good first start.

This addresses #2284

DO NOT SUBMIT this until:
 - [x] README is updated
 - [x] New Trillian Release has been pushed

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

**Comment by pphaneuf** [comment]
I'm @pphaneuf, and I support this PR. 😁

**Review by pphaneuf — APPROVED** [review]

**Comment by mhutchinson** [comment]
Moved this back to draft. Between the green build yesterday and this morning the goal posts have moved to a new city and I've got to play catch up with what the linter wants.

**Comment by mhutchinson** [comment]
It's building again. My rebasing skills have had a lot of practice from this PR.

**Review by pphaneuf — APPROVED** [review]

**Comment by mhutchinson** [comment]
@AlCutter or @Martin2112 - can haz LGTM please?

File: cloudbuild_tag.yaml [hunk 1]
```diff
@@ -48,12 +48,4 @@ steps:
...
-  waitFor: ["-"]
-- id: build_map_server
-  name: gcr.io/kaniko-project/executor
-  args:
-  - --dockerfile=examples/deployment/docker/map_server/Dockerfile
-  - --destination=gcr.io/${PROJECT_ID}/map_server:${TAG_NAME}
-  - --cache=true
-  - --cache-dir= # Cache is in Google Container Registry
-  waitFor: ["-"]
+  waitFor: ["-"]
```
Comment (Martin2112) [inline cloudbuild_tag.yaml:51]: Missing newline.

File: cmd/createtree/main_test.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func mustMarshalAny(p proto.Message) *any.Any {
 
 func TestCreateTree(t *testing.T) {
 	nonDefaultTree := proto.Clone(defaultTree).(*trillian.Tree)
-	nonDefaultTree.TreeType = trillian.TreeType_MAP
+	nonDefaultTree.TreeType = trillian.TreeType_LOG
 	nonDefaultTree.SignatureAlgorithm = sigpb.DigitallySigned_RSA
 	nonDefaultTree.DisplayName = "Llamas Map"
```
Comment (Martin2112) [inline cmd/createtree/main_test.go:70]: Tree name is still "Map".

File: experimental/batchmap/tmap.go [hunk 3]
```diff
@@ -16,9 +16,6 @@
 // verifiable data structures.
 package batchmap
 
-//go:generate go install github.com/apache/beam/sdks/go/cmd/starcgen
```
Comment (Martin2112) [inline experimental/batchmap/tmap.go:19]: Isn't this for the new stuff, is it still needed?

File: cloudbuild_tag.yaml (hunk 1 above)
Comment (mhutchinson) [reply cloudbuild_tag.yaml]: Done.

File: cmd/createtree/main_test.go (hunk 2 above)
Comment (mhutchinson) [reply cmd/createtree/main_test.go]: Updated.

File: experimental/batchmap/tmap.go (hunk 3 above)
Comment (mhutchinson) [reply experimental/batchmap/tmap.go]: Well spotted. This was supposed to be a temporary removal when I got into a bind with generated files. Somehow it jumped the sandbox and got into this PR. Reverted.

**Review by mhutchinson — COMMENTED** [review]
Ta

**Review by Martin2112 — APPROVED** [review]

**Pull request merged**